tests/test_DGEList.py

```diff
@@ -36,6 +36,11 @@ def test_setting_DGElist_counts():
     with pytest.raises(ValueError):
         dge_list().counts = np.array(1, 1, np.nan)
 
+def testing_setting_counts_and_values():
+    with pytest.raises(ValueError):
+        dge_list().counts = np.array(1, 1, 1)
+        dge_list().samples = np.array(1, 2)
```

Comment:
Two things:

1. You are actually raising a `ValueError` on your init of the numpy array and not on the assignment to counts or samples! To convince yourself type this into the interpreter:

```python
import numpy as np
np.array(1, 1 ,1)
```

2. The context `pytest.raises(ValueError)` is likely to only catch the first assertion and not all subsequent assertions. Could you double check? We might have this error elsewhere in our testing suite, as in, you are not the first to make it.

I was reading the `pytest` documentation and there is another, more terse, way of asserting exceptions:

```python
import pytest
pytest.raises(ExpectedException, "func(*args, **kwargs)")
```

Source: https://docs.pytest.org/en/latest/assert.html#assertions-about-expected-exceptions

Reply:
FYI I have been meaning to give a really hard pass at the `DGEList` object as it stands and construct a proper unit test suite instead of what we have.

This work would stand as a better example for others to copy.

I'll see if I can polish my work up this week.

Reply:
I think I'm wrong about that with statement claim. It likely is catching both exceptions...

Reply:
So, looking at this again now, I kinda wonder what I was doing in the first place.

So DGEList should raise a ValueError in the constructor, right? So I shouldn't even be making assignments. I should expect that creating an invalid DGEList object raises an error, but creating a valid one does not. I'll try to fix that now.

Reply:
Ah they appear to pass because you are raising type errors when you try to make those numpy arrays.

Reply:
So if I raised a type error with `np.array(1, 1, 1)`, then shouldn't the test on line 35 `np.array(-1, 1, 1)` fail for the same reason?